src/Workleap.DomainEventPropagation.Publishing/EventPropagationClient.cs

```diff
@@ -58,14 +67,51 @@ public async Task PublishDomainEventsAsync<T>(IEnumerable<T> domainEvents, Cance
         }
     }
 
-    private async Task SendDomainEventsAsync(DomainEventWrapperCollection domainEventWrappers, CancellationToken cancellationToken)
+    private Task SendDomainEventsAsync(DomainEventWrapperCollection domainEventWrappers, CancellationToken cancellationToken)
+    {
+        return domainEventWrappers.DomainSchema switch
+        {
+            EventSchema.EventGridEvent => this.SendEventGridEvents(domainEventWrappers, cancellationToken),
+            EventSchema.CloudEvent => this.SendCloudEvents(domainEventWrappers, cancellationToken),
+            _ => throw new NotSupportedException($"Event schema {domainEventWrappers.DomainSchema} is not supported"),
+        };
+    }
+
+    private Task SendEventGridEvents(
+        DomainEventWrapperCollection domainEventWrappers,
+        CancellationToken cancellationToken)
     {
+        var topicType = this._eventPropagationPublisherOptions.TopicType;
         var eventGridEvents = domainEventWrappers.Select(wrapper => new EventGridEvent(
             subject: wrapper.DomainEventName,
             eventType: wrapper.DomainEventName,
             dataVersion: DomainEventDefaultVersion,
             data: new BinaryData(wrapper.Data)));
 
-        await this._eventGridPublisherClient.SendEventsAsync(eventGridEvents, cancellationToken).ConfigureAwait(false);
+        return topicType switch
+        {
+            TopicType.Custom => this._eventGridPublisherClient.SendEventsAsync(eventGridEvents, cancellationToken),
+            TopicType.Namespace => throw new NotSupportedException("Cannot send EventGridEvents to a namespace topic"),
+            _ => throw new NotSupportedException($"Topic type {topicType} is not supported"),
+        };
+    }
+
+    private Task SendCloudEvents(
+        DomainEventWrapperCollection domainEventWrappers,
+        CancellationToken cancellationToken)
+    {
+        var topicType = this._eventPropagationPublisherOptions.TopicType;
+        var topicName = this._eventPropagationPublisherOptions.TopicName;
+        var cloudEvents = domainEventWrappers.Select(wrapper => new CloudEvent(
+            type: wrapper.DomainEventName,
+            source: this._eventPropagationPublisherOptions.TopicEndpoint,
+            jsonSerializableData: wrapper.Data));
+
+        return topicType switch
+        {
+            TopicType.Custom => this._eventGridPublisherClient.SendEventsAsync(cloudEvents, cancellationToken),
```

Comment:
Ici on enable d'envoyer des cloud events sur un custom topic, nice. L'affaire c'est que c'est pas handler à la réception avec le push model, so soit on modifie le code pour handler mais ça sort du scope de vos changements. Soit `NotSupportedException`?

Reply:
It is no longer a supported case and it throws a `NotSupportedException`